src/main/java/it/gov/pagopa/pu/fileshare/service/ingestion/IngestionFlowFileServiceImpl.java

```diff
@@ -55,4 +66,37 @@ public String uploadIngestionFlowFile(Long organizationId, IngestionFlowFileType
         ingestionFlowFileType, fileOrigin, organizationId, filePath)
       , accessToken);
   }
+
+  @Override
+  public FileResourceDTO downloadIngestionFlowFile(Long organizationId, Long ingestionFlowFileId, UserInfo user, String accessToken) {
+    userAuthorizationService.checkUserAuthorization(organizationId, user, accessToken);
+
+    IngestionFlowFile ingestionFlowFile = ingestionFlowFileClient.getIngestionFlowFile(ingestionFlowFileId, accessToken);
+
+    String filePath = getFilePath(organizationId, ingestionFlowFile);
+
+    InputStreamResource decryptedResource = fileStorerService.decryptFile(
+      filePath,
+      ingestionFlowFile.getFileName());
+
+    if (decryptedResource == null) {
+      log.error("downloadIngestionFlowFile - File [{}] could not be decrypted or was not found", ingestionFlowFile.getFileName());
+      throw new ResponseStatusException(HttpStatus.NOT_FOUND, "File could not be decrypted or was not found");
```

Comment:
Service layer should not handle HTTP information! here you should throw a custom exception, converted into HTTP status in the RestControllerAdvice